### PR TITLE
feat: basic metrics

### PR DESCRIPTION
**Pull request by moolen**
**What this PR does / why we need it**:
This PR Introduces a HTTP Server component that serves Prometheus metrics. We want metrics for this application to gain insights into the app and its integration with 3rd party APIs:
* do we successfully sync external secrets?
* if not what secrets, namespaces or backends are affected

This PR adds the `sync_calls` metric with labels `name|namespace|backend|status` that provide visibility for the above criteria.

### CODE REVIEW

File: lib/metrics-server.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    this._app.get('/metrics', (req, res) => {
+      res.set('Content-Type', Prometheus.register.contentType)
+      res.end(this._registry.metrics())
+    })
+  }
+
+  /**
+   * Start the metrics server: Listen on a TCP port and serve metrics over HTTP
+   */
+  async start () {
```
Comment (silasbw) [inline lib/metrics-server.js]: remove async

File: lib/metrics-server.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        resolve()
+      })
+      this._app.on('error', err => reject(err))
+    })
+  }
+
+  /**
+   * Stop the metrics server
+   */
+  async stop () {
```
Comment (silasbw) [inline lib/metrics-server.js]: remove async

**Comment by moolen** [comment]
thank you <3
I addressed the comments & rebased onto master

File: README.md [hunk 3]
```diff
@@ -53,6 +53,8 @@ The following table lists the configurable parameters of the `kubernetes-externa
 | Parameter                                 | Description                                                  | Default                                                 |
 | ----------------------------------------- | ------------------------------------------------------------ | ------------------------------------------------------- |
 | `env.AWS_REGION`                          | Set AWS_REGION in Deployment Pod                             | `us-west-2`                                             |
+| `env.LOG_LEVEL`                           | Set the application log level                                | `info`                                                  |
```
Comment (keweilu) [inline README.md]: Can you also update the charts README and values.yaml file with those two environment variables https://github.com/godaddy/kubernetes-external-secrets/tree/master/charts/kubernetes-external-secrets?

**Comment by moolen** [comment]
@silasbw @keweilu ptal

**Pull request merged**